packages/xod-project/src/patchPathUtils.js

```diff
@@ -137,12 +137,20 @@ export const getVariadicPath = n => `${PATCH_NODES_LIB_NAME}/variadic-${n}`;
 // utils for cast patches
 //
 
-const castTypeRegExp = new RegExp(
+const legacyCastTypeRegExp = new RegExp(
   `xod/core/cast-(${dataTypes.join('|')})-to-(${dataTypes.join('|')})$`
 );
+// TODO: When custom types will be added this should be generalized.
+//       Also, casting from/to generic types should be forbidden.
+const castTypeRegExp = new RegExp(
+  `xod/core/cast\\((${dataTypes.join('|')}),(${dataTypes.join('|')})\\)$`
```

Comment:
What about begin of the line in the regular expressions?

Reply:
Good point. Fixed.

Reply:
+ updated regexps for checking constant-*, variadic-* and internal terminals